### DISCUSSION TITLE
Homework 4 | Strings processing library | Inject and build validation tests

### DISCUSSION THREAD

**Discussion by PlungedInCode** [cpp-for-yourself/lectures-and-homeworks#74] [Homeworks Q&A] [answered]
Hi there. Just wanted to drop a quick note about a small hiccup I ran into while working on fourth task.

I noticed that the checker is trying to include the file <no_strings_attached/string_split.h>. Here output : 
Homework 4 | Strings processing library | Inject and build validation tests
`Status code 2`

`stderr`
```
/tmp/homework_checker/no_strings_attached_d9e74217afb88bbc68972dfe11ff752e/validation_tests/test_string_split.cpp:2:10: fatal error: no_strings_attached/string_split.h: No such file or directory
    2 | #include <no_strings_attached/string_split.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
gmake[2]: *** [validation_tests/CMakeFiles/validation_tests.dir/build.make:76: validation_tests/CMakeFiles/validation_tests.dir/test_string_split.cpp.o] Error 1
gmake[2]: *** Waiting for unfinished jobs....
/tmp/homework_checker/no_strings_attached_d9e74217afb88bbc68972dfe11ff752e/validation_tests/test_string_trim.cpp:2:10: fatal error: no_strings_attached/string_trim.h: No such file or directory
    2 | #include <no_strings_attached/string_trim.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
gmake[2]: *** [validation_tests/CMakeFiles/validation_tests.dir/build.make:90: validation_tests/CMakeFiles/validation_tests.dir/test_string_trim.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:1233: validation_tests/CMakeFiles/validation_tests.dir/all] Error 2
gmake: *** [Makefile:146: all] Error 2
```
`stdout`
```
-- CMAKE_BUILD_TYPE: Release
-- Configuring done (0.0s)
-- Generating done (0.0s)
-- Build files have been written to: /tmp/homework_checker/no_strings_attached_d9e74217afb88bbc68972dfe11ff752e/build
[  8%] Built target split_string
[ 17%] Built target gtest
[ 26%] Built target trim_string
[ 34%] Built target gmock
[ 43%] Built target gtest_main
[ 52%] Built target split_strings
[ 60%] Built target trim_strings
[ 69%] Built target split_string_test
[ 78%] Built target trim_string_test
[ 82%] Building CXX object validation_tests/CMakeFiles/validation_tests.dir/test_string_split.cpp.o
[ 91%] Built target gmock_main
[ 95%] Building CXX object validation_tests/CMakeFiles/validation_tests.dir/test_string_trim.cpp.o
```
but in lectures [(here)](https://github.com/cpp-for-yourself/lectures-and-homeworks/blob/main/homeworks/homework_4/homework.md) in the expected project structure, it was mentioned as "split_string.h."
 **Screenshot:**
![image](https://github.com/cpp-for-yourself/lectures-and-homeworks/assets/88050451/30f2d632-d971-4b4d-9729-ca08426da8c2)

In the tests, though, it's <no_strings_attached/string_split.h>[(here)](https://github.com/cpp-for-yourself/homework-definitions/blob/main/inject/homework_4/validation_tests/test_string_split.cpp). And also test files called `test_string_split.cpp`, `test_string_trim.cpp`  but in lectures they are `split_string_test.cpp`, `trim_string_test.cpp`

Just trying to keep things in line and avoid any confusion.

**Comment by niosus** [comment]
Thanks for pointing this out! I'm away from my laptop for a couple more days but will have a look afterwards. I try to keep things very consistent so anything that looks confusing is very welcome to be reported just like you did!

**Comment by niosus** [comment, marked as answer]
Ok, @PlungedInCode this should be now fixed as soon as PR https://github.com/cpp-for-yourself/lectures-and-homeworks/pull/76 is merged. I will mark the discussion resolved, but please continue it if the issue is still there.
